app.rb

```diff
@@ -18,194 +18,288 @@
   Dotenv.load
 end
 
-class Tint < Sinatra::Base
-  helpers Sinatra::Streaming
+module Tint
+  PROJECT_PATH = ENV["PROJECT_PATH"]
 
-  configure :development do
-    register Sinatra::Reloader
-  end
+  class App < Sinatra::Base
+    helpers Sinatra::Streaming
 
-  set :environment, Sprockets::Environment.new
+    configure :development do
+      register Sinatra::Reloader
+    end
 
-  environment.append_path "assets/stylesheets"
-  environment.css_compressor = :scss
+    set :environment, Sprockets::Environment.new
 
-  project_path = ENV['PROJECT_PATH']
+    environment.append_path "assets/stylesheets"
+    environment.css_compressor = :scss
 
-  get "/" do
-    erb :index
-  end
+    def project_path
+      PROJECT_PATH
+    end
 
-  get "/assets/*" do
-    env["PATH_INFO"].sub!("/assets", "")
-    settings.environment.call(env)
-  end
+    get "/" do
+      erb :index
+    end
 
-  get "/files" do
-    files = Dir.glob("#{project_path}/*")
-    erb :"files/index", locals: { files: files, root: project_path }
-  end
+    get "/assets/*" do
+      env["PATH_INFO"].sub!("/assets", "")
+      settings.environment.call(env)
+    end
 
-  get "/files/*" do
-    path = "#{project_path}/#{params['splat'].join('/')}"
-    if File.directory?(path)
-      files = Dir.glob("#{path}/*")
-      erb :"files/index", locals: { files: files, root: project_path }
-    else
-      is_text = FileMagic.open(:mime) do |magic|
-        magic.file(path).split('/').first == 'text'
-      end
+    get "/files" do
+      render_directory project_path
+    end
 
-      if is_text
-        has_content, has_frontmatter = detect_content_or_frontmatter(path)
-        if path.split(/\./).last.downcase == 'yml' || !has_content
-          data = YAML.safe_load(open(path))
-          erb :"files/yml", locals: { data: data, path: "/files" + path.gsub(project_path, "") }
+    get "/files/*" do
+      path = "#{project_path}/#{params['splat'].join('/')}"
+      file = Tint::File.new(path)
+
+      if file.directory?
+        render_directory path
+      elsif file.text?
+        if path.split(/\./).last.downcase == 'yml' || !file.content?
+          erb :"files/yml", layout: :files_layout, locals: {
+            data: file.frontmatter,
+            path: file.route
+          }
         else
-          wysiwyg = ['md', 'markdown'].include?(path.split(/\./).last.downcase)
-          frontmatter = has_frontmatter && YAML.safe_load(open(path))
+          frontmatter = file.frontmatter? && file.frontmatter
           stream do |out|
-            html = erb :"files/text", locals: { frontmatter: frontmatter, wysiwyg: wysiwyg, path: "/files" + path.gsub(project_path, "") }
+            html = erb :"files/text", layout: :files_layout, locals: {
+              frontmatter: frontmatter,
+              wysiwyg: file.markdown?,
```

Comment:
hmm.  This is ok.  Then instead of `#ext` above maybe `#yaml?` which checks for both yml and yaml ?